cli/azd/internal/grpcserver/compose_service.go

```diff
@@ -19,15 +21,20 @@ import (
 // composeService exposes features of the AZD composability model to the Extensions Framework layer.
 type composeService struct {
 	azdext.UnimplementedComposeServiceServer
-
 	lazyAzdContext *lazy.Lazy[*azdcontext.AzdContext]
+	env            *environment.Environment
+	envManager     environment.Manager
 }
 
 func NewComposeService(
 	lazyAzdContext *lazy.Lazy[*azdcontext.AzdContext],
+	env *environment.Environment,
+	envManager environment.Manager,
 ) azdext.ComposeServiceServer {
 	return &composeService{
 		lazyAzdContext: lazyAzdContext,
+		env:            env,
+		envManager:     envManager,
 	}
 }
```

Comment:
We have to be careful adding non-lazy dependencies to the gRPC server side services because the component initialization can fail. For example, try running the command in a fold that doesn't have an azd project or environment yet and this will likely fail.  

Take a look at some of the other gRPC services for examples to use lazy components.